src/tabs/tokens.jsx

```diff
@@ -3,97 +3,102 @@
 
   useEffect(() => {
-    if (address && !nfts && loginFrom) {
+    if (address && loginFrom) {
       setNfts([]);
-      (async () => {
-        if (loginFrom === 'metamask') {
-          const res = await fetch(`https://api.opensea.io/api/v1/assets?owner=${address}&limit=${50}`, {
-            headers: {
-              'X-API-KEY': '6a7ceb45f3c44c84be65779ad2907046',
-            },
-          });
-
-          const j = await res.json();
-          const {assets} = j;
-          setNfts(assets);
-        } else if (loginFrom === 'discord') {
-          let res = await fetch(`${tokensHost}/${address}`);
-          res = await res.json();
-          res = res.map(_nft => {
-            /** Modify the response recieved from the API-Backend to match standardise format */
-            _nft.image_preview_url = _nft.hash;
-            return _nft;
-          });
-          setNfts(res);
-        }
-      })();
+      if (loginFrom === 'metamask') {
+        console.log('fetching tokens');
+        (async () => {
+          const {chainName} = await new MetamaskWallet().getChainInfo();
+          if (chainName && chainName !== 'unknown') {
+            const res = await fetch(
+              `https://nft.webaverse.com/nft?chainName=${chainName}&owner=${address}`,
+            );
+            const nfts = await res.json();
+            setNfts(nfts);
+          } else {
+            console.log('Network not supported: ', chainName);
+          }
+        })();
+      } else if (loginFrom === 'discord') {
+        (async () => {
+          const res = await fetch(
+            `https://nft.webaverse.com/nft?chainName=sidechain&owner=${address}`,
+          );
+          const nfts = await res.json();
+          setNfts(nfts);
+        })();
+      }
     }
-  }, [address, nfts, loginFrom]);
+  }, [address, loginFrom]);
 
   useEffect(() => {
     if (nfts) {
-      for (const nft of nfts) {
-        if (!nftPreviews[nft.image_preview_url]) {
-          nftPreviews[nft.image_preview_url] = 'images/object.jpg';
-          if (loginFrom === 'metamask') {
-            fetch(nft.image_preview_url).then(response => response.blob())
-              .then(imageBlob => {
-                const imageObjectURL = URL.createObjectURL(imageBlob);
-                nftPreviews[nft.image_preview_url] = imageObjectURL;
-                setNftPreviews(nftPreviews);
-              });
-          } else if (loginFrom === 'discord') {
-            /** Will be switched after Previews-Merge */
-            // preview(nft.image_preview_url, nft.ext, 'png', 100, 100).then(res => {
-            //   const imageObjectURL = URL.createObjectURL(res.blob);
-            //   nftPreviews[nft.image_preview_url] = imageObjectURL;
-            //   setNftPreviews(nftPreviews);
-            // });
+      nfts.map(async nft => {
+        try {
+          if (!nft.metadata || !nft.metadata.image) {
+            throw new Error('No image use the default one');
```

Comment:
IMO trycatch should not be used to `goto`.

Reply:
What do you means by goto?

Here it tries to fetch the image for that NFT from the link which may or may not be on ipfs. If the request fails default image is shown.

Reply:
This is using `goto`-oriented programming. https://en.wikipedia.org/wiki/Goto

IMO cases like this can just be `if`.

Reply:
Updated the code